statvar_imports/us_bls/bls_ces_state/download_script/download_state_data.py

```diff
@@ -0,0 +1,290 @@
+                            writer.writerow(row)
+            if '(1)' in csv_file_path:
+                os.remove(csv_file_path)
+        os.remove(input_file_path)
+
+def download_series_data_from_url(folder_path, state_folder, \
+                                  county_folder, unresolved_county_folder, state_names):
+    state_url= "https://www.bls.gov/sae/additional-resources/list-of-published-state-and-metropolitan-area-series/"
+    service = Service(ChromeDriverManager().install()) 
+    driver = webdriver.Chrome(service=service)
+    driver.get("https://www.bls.gov/sae/additional-resources/list-of-published-state-and-metropolitan-area-series/home.htm")
+    html_content = driver.page_source
+    logging.fatal(f"Downloading series data from :{state_url}")
+    soup = BeautifulSoup(html_content, "html.parser")
+    if len(state_names)>0:
+        state_values = state_names
+    else:
+        state_values = [td.get_text() for td in soup.find_all('td')]
+    
+    state_name_values = write_to_config(state_values)
+    processed_state_names = ['-'.join(state.lower().split()) for state in state_values]
+    try:
+        for state in processed_state_names:
+                if(len(state)>1):
+                    full_state_url = f"{state_url}{state}.htm"
+                    driver.get(full_state_url)
+                    html_content = driver.page_source
+                    soup = BeautifulSoup(html_content, "html.parser")
+                    table = soup.find('table', class_='regular')
+                    rows = []
+                    for row in table.find_all('tr')[1:]:  
+                        columns = row.find_all('td')
+                        if columns:
+                            data = [column.get_text() for column in columns]
+                            rows.append(data)
+                    filename = state+".csv"
+                    filename = os.path.join(folder_path, filename)
+                    filename = filename.replace("-","_")
+                    headers = ['Series ID', 'Area','Industry','Datatype','Adjustment Method']
+                    with open(filename, mode='w', newline='') as file:
+                        writer = csv.writer(file)
+                        writer.writerow(headers)
+                        writer.writerows(rows)
+                    process_series_file(filename, state, state_folder, county_folder, unresolved_county_folder)
+                    os.remove(filename)
+        return state_name_values       
+    except WebDriverException as e:
+        logging.fatal(f"Error occurred with the web driver: {e}")
+
+    except ValueError as e:
+        logging.fatal(f"ValueError: {e}")
+
+    except Exception as e:
+        logging.fatal(f"one of State name from {state_values} does not exist, check again!")
+
+    finally:
+        driver.quit()
+        
+            
+        
+def process_series_file(input_csv_filename, state, state_folder, county_folder, unresolved_county_folder):
```

Comment:
can you please add a function block comment describing the input and what is changed for each  output file?